### ISSUE TITLE
Prettier removing a piece of code when formatting multi-line conditions

### ISSUE THREAD

**Issue by philcatterall** [prettier/eslint-plugin-prettier#532]

**Comment by fisker** [comment]
Transfered to eslint-plugin-prettier.

**Comment by fisker** [comment]
To fix this issue, we need find out what's the conflictting rule and extend it's fix range.

**Comment by philcatterall** [comment]
@kachkaev Hey Alex, out of curiosity (even though I've now abided by Prettier's "opinion" :-), I was fiddling with the .eslintrc settings, bearing in mind your feedback, playing with different settings. After several hours of googling, and experimentation, I think I'm now more confused than ever, given the high number of combinations of extends, parseroptions, settings, plugins and rules !!!. So I hope you don't mind me asking, but could you post for me your recommended settings ? I'm using vsCode and mostly coding in Typescript, with a bit of Javascript and some JSON files. Also for some projects using React. Any guidance you could give me would be very welcome. Many thanks, Phil

**Comment by kachkaev** [comment]
Yeah there are a few ways to integrate ESLint and Prettier for historical reasons, which can be confusing indeed:
https://prettier.io/docs/en/related-projects.html#eslint-integrations

You can find some thoughts on this topic in this answer: https://github.com/prettier/prettier/discussions/13925#discussioncomment-4276876.

As of project examples, I can’t recall something that would be easy to copy-paste. My [`.prettierrc.cjs`](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/.prettierrc.cjs) and [`.eslintrc.cjs`](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/.eslintrc.cjs) often depend on custom packages with rules which is a bit bespoke but is helpful for maintaining several projects. You can [explore these packages](https://github.com/kachkaev/routine-npm-packages/tree/main/packages) but please keep in mind that these are not industry standards, just my own approach to linting config. When using VSCode, I enable Prettier and ESLint extensions and make sure that they both run on save. Prettier runs second. Editor extensions call both tools similarly [to how you can do this from CLI](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/package.json#L11-L21). CLI scripts [can be launched from GitHub Workflows](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/.github/workflows/ci.yml#L52-L62), thus helping your team to spot any regressions in linting.


This is a bit offtopic, so please feel free to create a discussion and tag me there if you need more help.

**Comment by philcatterall** [comment]
> Yeah there are a few ways to integrate ESLint and Prettier for historical reasons, which can be confusing indeed: https://prettier.io/docs/en/related-projects.html#eslint-integrations
> 
> You can find some thoughts on this topic in this answer: [prettier/prettier#13925 (comment)](https://github.com/prettier/prettier/discussions/13925#discussioncomment-4276876).
> 
> As of project examples, I can’t recall anything that would be easy to copy-paste. My [`.prettierrc.cjs`](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/.prettierrc.cjs) and [`.eslintrc.cjs`](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/.eslintrc.cjs) files often depend on custom packages with rules which is a bit bespoke but is helpful for maintaining several projects. You can [explore these packages](https://github.com/kachkaev/routine-npm-packages/tree/main/packages) but please keep in mind that these are not a community standard, just my own approach to linting side projects. When using VSCode, I enable Prettier and ESLint extensions and make sure that they both run on save. Prettier runs second automatically (the order is important here). Editor extensions call both tools similarly [to how you can do this from CLI](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/package.json#L11-L21). CLI scripts [can be launched from GitHub Workflows](https://github.com/kachkaev/njt/blob/b153942f8d85287489bf3d9242eaa07443671e92/.github/workflows/ci.yml#L52-L62), thus helping your team to spot any regressions in linting.
> 
> This is a bit offtopic, so please feel free to create a discussion and tag me there if you need more help.

Hi Alex, many thanks, and yes sorry, it is a bit off-topic, but your reply is sincerely appreciated. I will follow the rules from now and continue in discussions if I need, (embracing fully Prettier's ethos !! Cheers
